src/Interpreters/ExpressionAnalyzer.cpp

```diff
@@ -177,20 +176,12 @@ bool checkPositionalArguments(ASTPtr & argument, const ASTSelectQuery * select_q
             }
         }
         else
-            positional = false;
-    }
-    else if (const auto * ast_function = typeid_cast<const ASTFunction *>(argument.get()))
-    {
-        if (ast_function->arguments)
-        {
-            for (auto & arg : ast_function->arguments->children)
-                positional &= checkPositionalArguments(arg, select_query, expression);
-        }
+            return false;
```

Comment:
It would be better to use [return early pattern](https://szymonkrajewski.pl/why-should-you-return-early/) here to get rid of some nested branches and make code more readable, but it's minor remark, feel free to ignore it.